jsonrpc/namespaces/eth/filters/filter_system.go

```diff
@@ -0,0 +1,340 @@
+// Copyright 2021 Evmos Foundation
+// This file is part of Evmos' Ethermint library.
+//
+// The Ethermint library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Ethermint library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Ethermint library. If not, see https://github.com/evmos/ethermint/blob/main/LICENSE
+package filters
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"cosmossdk.io/log"
+	tmjson "github.com/cometbft/cometbft/libs/json"
+	tmquery "github.com/cometbft/cometbft/libs/pubsub/query"
+	coretypes "github.com/cometbft/cometbft/rpc/core/types"
+	rpcclient "github.com/cometbft/cometbft/rpc/jsonrpc/client"
+	tmtypes "github.com/cometbft/cometbft/types"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/eth/filters"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/initia-labs/minievm/jsonrpc/pubsub"
+	evmtypes "github.com/initia-labs/minievm/x/evm/types"
+)
+
+var (
+	txEvents     = tmtypes.QueryForEvent(tmtypes.EventTx).String()
+	headerEvents = tmtypes.QueryForEvent(tmtypes.EventNewBlockHeader).String()
+	blockEvents  = tmtypes.QueryForEvent(tmtypes.EventNewBlock).String()
+	evmEvents    = tmquery.MustCompile(
+		fmt.Sprintf("%s='%s' AND %s.%s='%s'",
+			tmtypes.EventTypeKey,
+			tmtypes.EventTx,
+			sdk.EventTypeMessage,
+			sdk.AttributeKeyModule, evmtypes.ModuleName),
+	).String()
+)
+
+// EventSystem creates subscriptions, processes events and broadcasts them to the
+// subscription which match the subscription criteria.
+type EventSystem struct {
+	logger     log.Logger
+	ctx        context.Context
+	tmWSClient *rpcclient.WSClient
```

Comment:
@jasonsong0 
Could you please change `tmWSClient` to `cometWSClient` in this file?